com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/steps/PollStartAppStatusExecution.java

```diff
@@ -139,38 +139,36 @@ private void showInstancesStatus(ExecutionWrapper execution, String appName, Lis
             for (InstanceInfo instance : instances) {
                 final String state = instance.getState()
                     .toString();
-                final Integer stateCount = stateCounts.get(state);
-                stateCounts.put(state, (stateCount == null) ? 1 : (stateCount + 1));
+                stateCounts.compute(state, (k, stateCount) -> stateCount == null ? 1 : stateCount + 1);
             }
         }
 
         // Compose state strings
-        List<String> stateStrings = new ArrayList<>();
-        for (Map.Entry<String, Integer> sc : stateCounts.entrySet()) {
-            stateStrings.add(format("{0} {1}", sc.getValue(), sc.getKey()
-                .toLowerCase()));
-        }
+        String states = stateCounts.entrySet()
+            .stream()
+            .map(this::formatStateString)
+            .collect(Collectors.joining(","));
 
         // Print message
-        String message = format(Messages.APPLICATION_0_X_OF_Y_INSTANCES_RUNNING, appName, runningInstances, expectedInstances,
-            String.join(",", stateStrings));
+        String message = format(Messages.APPLICATION_0_X_OF_Y_INSTANCES_RUNNING, appName, runningInstances, expectedInstances, states);
         execution.getStepLogger()
             .debug(message);
     }
 
+    private String formatStateString(Map.Entry<String, Integer> entry) {
+        return format("{0} {1}", entry.getValue(), entry.getKey()
+            .toLowerCase());
+    }
+
     private static List<InstanceInfo> getApplicationInstances(CloudControllerClient client, CloudApplication app) {
         InstancesInfo instancesInfo = client.getApplicationInstances(app);
         return (instancesInfo != null) ? instancesInfo.getInstances() : null;
     }
 
     private static int getInstanceCount(List<InstanceInfo> instances, InstanceState state) {
-        int count = 0;
-        for (InstanceInfo instance : instances) {
-            if (instance.getState()
-                .equals(state)) {
-                count++;
-            }
-        }
-        return count;
+        return (int) instances.stream()
+            .filter(instance -> instance.getState()
+                .equals(state))
+            .count();

```

Comment:
Another possible thing is to collect a list and then get the size(). Or make the method return long instead ?